File: evd_authoring_app/src/model/evd_script/program/tasks/close_gripper.js
import { Task } from '../task';
import { Gripper } from '../primitive';
import { NodeParser } from '../../utility_functions';


export class CloseGripper extends Task {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'close-gripper';
    }

    static fullTypeString() {
        return Task.fullTypeString() + CloseGripper.typeString();
    }

    constructor(position=100, effort=100, speed=100, thingUuid=null, type='', name='', uuid=null, parent=null, appendType=true, primitives=null) {

        if (primitives !== null) {
            primitives = [
                Gripper(
                    thingUuid= thingUuid,
                    position= position,
                    effort= effort,
                    speed= speed
                )
            ];
        }

        super(
            type= (appendType) ? 'close-gripper.'+type : type,
            name= name,
            uuid= uuid,
            parent= parent,
            appendType= appendType,
            primitives= primitives
        );
    }

    static fromDict(dct) {
        return CloseGripper(
            type= dct.type,
            name= dct.name,
            uuid= dct.uuid,
            appendType= false,
            primitives= dct.primitives.map(p => NodeParser(p))
        );
    }
}

File: evd_ros_backend/evd_ros_core/src/evd_script/program_nodes/skills/close_gripper.py
'''
Simplified parameterization for gripper primitive to fully closed
'''

from ...data_nodes.skill_argument import SkillArgument
from ...data_nodes.thing import Thing
from ..skill import Skill
from ..primitives import Gripper


def CloseGripper(name=None, uuid=None, parent=None, editable=True, deleteable=False, description=''):
 
    thing_uuid_arg = SkillArgument(
                        parameter_key='thing_uuid', 
                        name='thing_uuid', 
                        parameter_type=Thing.full_type_string())
    arguments = [thing_uuid_arg]

    primitives = [
        Gripper(
            thing_uuid=thing_uuid_arg.temporary_value,
            semantic=Gripper.SEMANTIC_GRASPING,
            position=100,
            editable=editable,
            deleteable=False)
    ]

    return Skill(
        name=name if name != None else 'Close Gripper',
        uuid=uuid,
        parent=parent,
        editable=editable,
        deleteable=deleteable,
        description=description,
        arguments=arguments,
        primitives=primitives)